src/corvus.c

```diff
@@ -134,30 +138,58 @@ int config_add(char *name, char *value)
             memcpy(config.requirepass, value, strlen(value));
         }
     } else if (strcmp(name, "node") == 0) {
-        cv_free(config.node.addr);
-        memset(&config.node, 0, sizeof(config.node));
-
+    	struct address *addr = NULL;
+    	int addr_cnt = 0;
         char *p = strtok(value, ",");
         while (p) {
-            config.node.addr = cv_realloc(config.node.addr,
-                    sizeof(struct address) * (config.node.len + 1));
-
-            if (socket_parse_ip(p, &config.node.addr[config.node.len]) == -1) {
-                cv_free(config.node.addr);
-                return -1;
+        	addr = cv_realloc(addr, sizeof(struct address) * (addr_cnt + 1));
+            if (socket_parse_ip(p, &addr[addr_cnt]) == -1) {
+                cv_free(addr);
+                return CORVUS_ERR;
             }
-
-            config.node.len++;
+            addr_cnt++;
             p = strtok(NULL, ",");
         }
+		{
+			struct node_conf *newnode = cv_calloc(1, sizeof(struct node_conf));
+			memset(newnode, 0, sizeof(struct node_conf));
+			newnode->addr = addr;
+			newnode->len = addr_cnt;
+			newnode->refcount = 1;
+			struct node_conf *oldnode = config.node;
+			config.node = newnode;
```

Comment:
node_dec_ref has to be reserved to make things simple if you just take it as "i dont want to use it anymore".

lock_config has been removed and it considered to be a config cmd lock. but no logic cannot be multi-thead currently i just remove it now.

i add same lock in pointer swapping.